In [1]:
import pandas as pd
import numpy as np

In [3]:
# 1 
sinasc = pd.read_csv('SINASC_RO_2019.csv')
sinasc.shape

(27028, 69)

In [5]:
sinasc.drop_duplicates()
sinasc.shape

# não há dados duplicados na base em estudo

(27028, 69)

In [6]:
# 2
sinasc.isna().sum()

ORIGEM          0
CODESTAB      115
CODMUNNASC      0
LOCNASC         0
IDADEMAE        0
             ... 
munResUf        0
munResLat       1
munResLon       1
munResAlt       1
munResArea      1
Length: 69, dtype: int64

In [8]:
# 3
base_sinasc = sinasc[['LOCNASC', 'IDADEMAE', 'ESTCIVMAE', 'ESCMAE', 'QTDFILVIVO', 
    'GESTACAO', 'GRAVIDEZ', 'CONSULTAS', 'APGAR5']]

base_sinasc
base_sinasc.isna().sum()

LOCNASC          0
IDADEMAE         0
ESTCIVMAE      317
ESCMAE         312
QTDFILVIVO    1573
GESTACAO      1232
GRAVIDEZ        79
CONSULTAS        0
APGAR5         103
dtype: int64

In [10]:
# 4 
print(base_sinasc.shape)

(27028, 9)


In [11]:
base_sinasc.dropna(subset=['APGAR5'], inplace = True)
print(base_sinasc.shape)

(26925, 9)


C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\3733228892.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.dropna(subset=['APGAR5'], inplace = True)


In [12]:
# 5
base_sinasc.isna().sum()

LOCNASC          0
IDADEMAE         0
ESTCIVMAE      315
ESCMAE         310
QTDFILVIVO    1566
GESTACAO      1216
GRAVIDEZ        76
CONSULTAS        0
APGAR5           0
dtype: int64

In [13]:
base_sinasc.ESTCIVMAE.fillna(9,inplace=True)
base_sinasc.CONSULTAS.fillna(9,inplace=True)

base_sinasc.isna().sum()

C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\2992306254.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.ESTCIVMAE.fillna(9,inplace=True)
C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\2992306254.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.CONSULTAS.fillna(9,inplace=True)


LOCNASC          0
IDADEMAE         0
ESTCIVMAE        0
ESCMAE         310
QTDFILVIVO    1566
GESTACAO      1216
GRAVIDEZ        76
CONSULTAS        0
APGAR5           0
dtype: int64

In [15]:
# 6
base_sinasc.QTDFILVIVO.fillna(0,inplace=True)
base_sinasc.isna().sum()

C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\2958002090.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.QTDFILVIVO.fillna(0,inplace=True)


LOCNASC          0
IDADEMAE         0
ESTCIVMAE        0
ESCMAE         310
QTDFILVIVO       0
GESTACAO      1216
GRAVIDEZ        76
CONSULTAS        0
APGAR5           0
dtype: int64

In [17]:
# 7
base_sinasc.ESCMAE.isna().sum()

310

In [18]:
base_sinasc.GESTACAO.isna().sum()

1216

In [19]:
base_sinasc.GRAVIDEZ.isna().sum()

76

In [20]:
base_sinasc.ESCMAE.value_counts()

8 a 11 anos        15581
12 anos ou mais     5757
4 a 7 anos          4727
1 a 3 anos           510
Nenhuma               40
Name: ESCMAE, dtype: int64

In [22]:
base_sinasc[['ESCMAE', 'GESTACAO', 'GRAVIDEZ']].value_counts()

ESCMAE           GESTACAO             GRAVIDEZ       
8 a 11 anos      37 a 41 semanas      Única              13098
12 anos ou mais  37 a 41 semanas      Única               4665
4 a 7 anos       37 a 41 semanas      Única               3846
8 a 11 anos      32 a 36 semanas      Única               1070
4 a 7 anos       32 a 36 semanas      Única                443
8 a 11 anos      42 semanas e mais    Única                436
12 anos ou mais  32 a 36 semanas      Única                399
1 a 3 anos       37 a 41 semanas      Única                397
4 a 7 anos       42 semanas e mais    Única                143
8 a 11 anos      32 a 36 semanas      Dupla                120
                 37 a 41 semanas      Dupla                112
12 anos ou mais  42 semanas e mais    Única                 96
8 a 11 anos      28 a 31 semanas      Única                 71
12 anos ou mais  32 a 36 semanas      Dupla                 56
1 a 3 anos       32 a 36 semanas      Única                 55
1

escolaridade da mãe, **310** dados vazios, 9: ignorado <br>
gestação, **1216** dados vazios, dados faltantes acrescidos ao resultado mais recorrente da base<br>
gravidez, **76** dados faltantes, dados faltantes acrescidos ao resultado mais recorrente da base <br>

In [23]:
base_sinasc.GESTACAO.fillna('37 a 41 semanas', inplace=True)
base_sinasc.isna().sum()

C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\278087445.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.GESTACAO.fillna('37 a 41 semanas', inplace=True)


LOCNASC         0
IDADEMAE        0
ESTCIVMAE       0
ESCMAE        310
QTDFILVIVO      0
GESTACAO        0
GRAVIDEZ       76
CONSULTAS       0
APGAR5          0
dtype: int64

In [24]:
base_sinasc.ESCMAE.fillna('Ignorado', inplace=True)
base_sinasc.ESCMAE.isna().sum()

C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\3620485530.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.ESCMAE.fillna('Ignorado', inplace=True)


0

In [25]:
base_sinasc.GRAVIDEZ.fillna('Ignorado',inplace=True)
base_sinasc.isna().sum()

C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\2436312094.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.GRAVIDEZ.fillna('Ignorado',inplace=True)


LOCNASC       0
IDADEMAE      0
ESTCIVMAE     0
ESCMAE        0
QTDFILVIVO    0
GESTACAO      0
GRAVIDEZ      0
CONSULTAS     0
APGAR5        0
dtype: int64

In [32]:
# 8
base_sinasc.loc[(base_sinasc.APGAR5 >= 8) & (base_sinasc.APGAR5 <= 10), 'CLASSAPGAR5'] = 'normal'
base_sinasc.loc[(base_sinasc.APGAR5 == 7) | (base_sinasc.APGAR5 == 6), 'CLASSAPGAR5'] = 'asfixia leve'
base_sinasc.loc[(base_sinasc.APGAR5 == 4) | (base_sinasc.APGAR5 == 5), 'CLASSAPGAR5'] = 'asfixia moderada'
base_sinasc.loc[(base_sinasc.APGAR5 >= 1) & (base_sinasc.APGAR5 <= 3), 'CLASSAPGAR5'] = 'asfixia grave'

In [33]:
base_sinasc.CLASSAPGAR5.value_counts()

normal              26480
asfixia leve          320
asfixia moderada       68
asfixia grave          57
Name: CLASSAPGAR5, dtype: int64

In [37]:
# 9
base_sinasc.rename(columns={'LOCNASC': 'loc_nasc', 'IDADEMAE': 'idade_mae', 'ESTCIVMAE': 'est_civ_mae', 'QTDFILVIVO': 'qtd_fil_vivo',
                           'GESTACAO': 'gestacao', 'GRAVIDEZ': 'gravidez', 'CONSULTAS': 'consultas', 'APGAR5': 'apgar5',
                            'CLASSAPGAR5': 'class_apgar5'}, inplace=True)

C:\Users\Usuario\AppData\Local\Temp\ipykernel_1796\4096901877.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  base_sinasc.rename(columns={'LOCNASC': 'loc_nasc', 'IDADEMAE': 'idade_mae', 'ESTCIVMAE': 'est_civ_mae', 'QTDFILVIVO': 'qtd_fil_vivo',


In [38]:
base_sinasc.columns

Index(['loc_nasc', 'idade_mae', 'est_civ_mae', 'ESCMAE', 'qtd_fil_vivo',
       'gestacao', 'gravidez', 'consultas', 'apgar5', 'class_apgar5'],
      dtype='object')

In [39]:
base_sinasc

loc_nasc  idade_mae  est_civ_mae           ESCMAE  qtd_fil_vivo  \
0             1         19          5.0      8 a 11 anos           0.0   
1             1         29          2.0      8 a 11 anos           1.0   
2             1         37          9.0      8 a 11 anos           2.0   
3             1         30          5.0  12 anos ou mais           0.0   
4             1         30          2.0      8 a 11 anos           1.0   
...         ...        ...          ...              ...           ...   
27023         1         32          2.0  12 anos ou mais           1.0   
27024         1         19          1.0      8 a 11 anos           0.0   
27025         1         24          2.0      8 a 11 anos           0.0   
27026         1         21          2.0      8 a 11 anos           1.0   
27027         1         18          5.0      8 a 11 anos           1.0   

              gestacao gravidez  consultas  apgar5 class_apgar5  
0      37 a 41 semanas    Única          4    10.0       normal  
1      37 a 41 semanas    Única          4     9.0       normal  
2      37 a 41 semanas    Única          4    10.0       normal  
3      37 a 41 semanas    Única          3    10.0       normal  
4      37 a 41 semanas    Única          4    10.0       normal  
...                ...      ...        ...     ...          ...  
27023  32 a 36 semanas    Única          4     9.0       normal  
27024  37 a 41 semanas    Única          4     9.0       normal  
27025  37 a 41 semanas    Única          4    10.0       normal  
27026  32 a 36 semanas    Única          4     9.0       normal  
27027  37 a 41 semanas    Única          4     8.0       normal  

[26925 rows x 10 columns]